sips/sip-015/sip-015-network-upgrade.md

```diff
@@ -0,0 +1,1582 @@
+
+This SIP proposes a set of updates to three major areas of the Stacks blockchain:
+
+* **Stacking**, the Proof-of-Transfer (PoX) consensus algorithm as implemented
+in the Stacks chain and originally proposed in [SIP-007](./sip-007-stacking-consensus.md). 
+The proposed changes improve the user and developer experience of participating
+in Stacking, and add support for new behaviors and PoX reward address types.
+
+* **Clarity**, the smart contract language supported on the Stacks blockchain.
+The proposed changes fix bugs and add new native functions and global variables
+that improve its support for interacting with off-chain services and
+blockchains, especially Bitcoin.
+
+* **Block Validation**, the procedure by which blocks are determined to be acceptable to the Stacks
+  blockchain.  The proposed changes address bugs in the implementation that are
+not specified by any prior SIP, but which cannot be changed without a
+coordinated network-wide upgrade.  In addition, the proposed changes address
+potential consensus challenges for PoX-related chain reorganizations which were
+not known at the time SIP-007 was written.  Finally, new variations of existing
+transactions are proposed to in order to better support Stacking, to support
+multiple Clarity versions, and to support decentralized mining pools.
+
+In addition to proposing these changes, this SIP also outlines an approach for
+implementing these changes in the Stacks blockchain.
+
+# Introduction
+
+This SIP condenses lessons learned from running the Stacks blockchain in
+production to date.  Since the system was launched in January 2021, several
+shortcomings have been discovered which can only be addressed through a
+coordinated, backwards-compatible network upgrade.  This SIP enumerates these
+shortcomings and proposes solutions.
+
+Broadly speaking, the shortcomings of the system fall into one of three
+categories: Stacking, the Clarity language, and block validation.
+
+Concerning Stacking, the current PoX implementation in the Stacks blockchain has several
+shortcomings for developers and users:
+
+1. After unlocking, Stackers **must** wait one cycle before Stacking
+   again. This is an implementation consequence, not a requirement of
+   the algorithm itself.
+
+2. If users fail to qualify during a cycle, their funds remain locked
+   for the cycle regardless.
+
+3. PoX cannot support advanced behaviors that could be used for
+   different kinds of smart contracts: changing the PoX reward
+   address, increasing or decreasing lock amounts, etc. These features
+   would enable new kinds of applications that build on top of PoX.
+
+4. PoX will automatically sunset after a pre-defined number of Bitcoin blocks
+   pass, regardless of how popular or useful it is.
+
+This SIP proposes the creation of a new `pox-2` smart contract to implement
+Stacking which addresses the above problems.  A migration procedure is presented
+for transitionin the system from `pox` to `pox-2`.
```

Comment:
I realise the word 'backwards-compatible' are in the section above but it might be worth to include in this paragraph about the pox-2 migration that the contracts already deployed using Clarity 1 and pox will be accessible and functioning. 

In short summarising the points in the section 'Backwards Compatibility' in 'lines 1366 - 1383

Reply:
Yup, can do.